### ISSUE TITLE
namespace-id and container_name not registered in the default installation

### ISSUE THREAD

**Issue by mgenov** [Stackdriver/kubernetes-configs#47]
I'm not sure whether this is a role of the stackdriver-metadata-agent but this information is required by the java logging client but it's not available in the workloads. 

The k8s cluster is configured with Stackdriver Kubernetes Engine Monitoring and Workload Identity enabled. All agents are running in the kube-system:

```
stackdriver-metadata-agent-cluster-level-74785fffdd-79b6v        1/1     Running   0          3h46m
 ```
The logs show no errors but the information regarding container_name and namespace_id is not available inside the containers: 

```
root@workload-identity-test:/# curl "http://metadata.google.internal/computeMetadata/v1/instance/attributes/"  -H "Metadata-Flavor: Google"
cluster-name
root@workload-identity-test:/#
```

e.g only cluster-name is available but the google-cloud-logging libraries are looking  and for namespace-id and container_name.

**Comment by mgenov** [comment]
Is anyone aware of this issue?

**Comment by igorpeshansky** [comment]
Hi. The metadata agent does not provide a metadata service. Rather, it queries the GCE metadata server to ingest metadata into Stackdriver.

That said, which `google-cloud-logging` library are you referring to? I couldn't find any mention of `container_name` or `namespace_id` in either [Python](https://github.com/googleapis/google-cloud-python/tree/master/logging), or [Ruby](https://github.com/googleapis/google-cloud-ruby/tree/master/google-cloud-logging), or [Java](https://github.com/googleapis/google-cloud-java/tree/master/google-cloud-clients/google-cloud-logging) libraries (the libraries for C#, Node.js, PHP, or Go don't use the name `google-cloud-logging`, see [this](https://cloud.google.com/logging/docs/reference/libraries)). In any case, you're probably better off opening an issue in the repository for those libraries.

**Comment by mgenov** [comment]
It's in the Java client library google-cloud-logging.

https://github.com/googleapis/google-cloud-java/blob/master/google-cloud-clients/google-cloud-logging/src/main/java/com/google/cloud/logging/MonitoredResourceUtil.java

**Comment by igorpeshansky** [comment]
Ah, I see. I did see the enum, but didn't notice that it mapped to `com.google.cloud.MetadataConfig`, which now lives [in another repo](https://github.com/googleapis/java-core/blob/master/google-cloud-core/src/main/java/com/google/cloud/MetadataConfig.java).

So, the bad news is that this functionality was added in https://github.com/googleapis/google-cloud-java/pull/3887, which is misguided, and could never work: there is no way for the GCE metadata server to serve the correct container name and namespace id to all containers, because the granularity of that metadata is per-node. Maybe with workload identity it becomes possible, but this change pre-dates that, as far as I can see.

Your best bet is to raise this issue in the [google-cloud-java](https://github.com/googleapis/google-cloud-java) repo (this becomes more complicated as the erroneous code is now split across that and the [java-core](https://github.com/googleapis/java-core) repo). @chingor13 may have some suggestions on how to manage the issues across the two repos. @meltsufin FYI.
